std/asyncio/net/Socket.hx

```diff
@@ -0,0 +1,65 @@
+package asyncio.net;
+
+import asyncio.net.SocketOption.SocketOptionKind;
+import haxe.NoData;
+import haxe.io.Bytes;
+import haxe.Callback;
+import haxe.errors.NotImplemented;
+
+class Socket implements IDuplex {
```

Comment:
@nadako made a good point – TCP and IPC sockets should be separate, let this be a base class for `TcpSocket` and `IpcSocket`.

Reply:
I'm not sure about that anymore. They share the API and can be distinguished by connection addresses. What would be the benefit of separating them like that?

Reply:
You could probably simplify the constructors and/or thngs like `connect`.